pkg/drivers/powerstore.go

```diff
@@ -44,6 +63,64 @@ func PrecheckPowerStore(ctx context.Context, cr *csmv1.ContainerStorageModule, o
 	}
 	secrets := []string{config}
 
+	for _, name := range secrets {
+		found := &corev1.Secret{}
+		err := ct.Get(ctx, types.NamespacedName{Name: name, Namespace: cr.GetNamespace()}, found)
+		if err != nil {
+			log.Error(err, "Failed query for secret ", name)
+			if errors.IsNotFound(err) {
+				return fmt.Errorf("failed to find secret %s", name)
+			}
+		}
+	}
+
 	log.Debugw("preCheck", "secrets", len(secrets))
 	return nil
 }
+
+// ModifyPowerstoreCR -
+func ModifyPowerstoreCR(YamlString string, cr csmv1.ContainerStorageModule,  file_type string) string {
+	// Parameters to initialise CR values
+	nodeprefix := ""
+	fcportfilter := ""
+	nfsacls := ""
+	healthmonitor_controller := ""
+	chap := ""
+	healthmonitor_node := ""
```

Comment:
Same comment as above for all the variables declared from Line No 84-89

Reply:
taken care